### PR TITLE
Fix misplaced negative sign on a currency visualization in Dashboard

### LINKED ISSUE #34242: '-' sign misplaced in shortened visualization of currency type
**Describe the bug**
In cards with a Number visualization, the negative sign of currency type numbers is incorrectly positioned when the card has to displays the shortened version of the number.

**Logs**
n/a

**To Reproduce**
1. Create a SQL question `SELECT -500000`, set visualization type to Number, style to Currency.
2. Save, and add it to a dashboard.
3. When there is enough space the number is displayed as `-$500,000`.
4. Decrease the size of the card, and the number will change to`$-500.0k`, the currency symbol and '-' sign are replaced.

**Expected behavior**
It should look like `-$500.0k`.

**Screenshots**

**Severity**
Reported by a customer.

**Additional context**
n/a

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1",
    "os.name": "Linux",
    "os.version": "5.10.186-179.751.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "bigquery-cloud-sdk",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.4"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-09-18",
      "tag": "v1.47.2",
      "branch": "release-x.47.x",
      "hash": "536c24c"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### PR DESCRIPTION
**Pull request by ibedwi**
Closes https://github.com/metabase/metabase/issues/34242

#### Root cause

In the `formatNumberCompact` function, the `currency` concatenated directly on a string version of a number.

#### How to solve

In this PR, the value's sign concatenated after the `currency` concatenated on the string version of a number.

### How to verify

Describe the steps to verify that the changes are working as expected.

For negative value:
1. Create a SQL question `SELECT -500000`, set visualization type to Number, style to Currency.
2. Save, and add it to a dashboard.
3. When there is enough space the number is displayed as `-$500,000`.
4. Decrease the size of the card, and the number will change to `$-500.0k`, the currency symbol and '-' sign are replaced.

For positive value
1. Create a SQL question `SELECT 600000`, set visualization type to Number, style to Currency.
2. Save, and add it to a dashboard.
3. When there is enough space the number is displayed as `$600,000`.
4. Decrease the size of the card, and the number will change to `$600.0k`, the currency symbol and '-' sign are replaced.

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

#### Before

CleanShot 2023-11-11 at 16 36 26@2x

#### After

CleanShot 2023-11-11 at 16 33 40@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/lib/formatting/numbers.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { formatNumber, numberFormatterForOptions } from "./numbers";
+
+describe("formatNumber", () => {
+  it("should the correct currency format (metabase#34242)", () => {
```
Comment (ibedwi) [inline frontend/src/metabase/lib/formatting/numbers.unit.spec.js:4]: Added the issue's reference as suggested by @kamilmielnik [on my previous PR](https://github.com/metabase/metabase/pull/34680#discussion_r1368193794)

**Review by alxnddr — APPROVED** [review]
Looks good! Thank you for the contribution!
We'll check CI here https://github.com/metabase/metabase/pull/35660 and if everything is good, merge the PR

**Comment by ibedwi** [comment]
> Looks good! Thank you for the contribution! We'll check CI here #35660 and if everything is good, merge the PR

Hi @alxnddr , I think there's a failing unit test. I've updated the unit test and run the test locally. Thanks!

**Comment by alxnddr** [comment]
Merging since FE CI jobs are passing https://github.com/metabase/metabase/pull/35660

**Pull request merged**